apps/evm/lib/evm/builtin.ex

```diff
@@ -138,14 +134,111 @@ defmodule EVM.Builtin do
           {EVM.Gas.t(), EVM.SubState.t(), EVM.ExecEnv.t(), EVM.VM.output()}
   def run_id(gas, exec_env) do
     data = exec_env.data
-    used_gas = @g_identity_base + @g_identity_byte * MathHelper.bits_to_words(byte_size(data))
+    required_gas = @g_identity_base + @g_identity_byte * MathHelper.bits_to_words(byte_size(data))
 
-    if used_gas <= gas do
-      remaining_gas = gas - used_gas
+    if required_gas <= gas do
+      remaining_gas = gas - required_gas
 
       {remaining_gas, %EVM.SubState{}, exec_env, data}
     else
       {0, %EVM.SubState{}, exec_env, :failed}
     end
   end
+
+  @spec exp_mod(EVM.Gas.t(), EVM.ExecEnv.t()) ::
+          {EVM.Gas.t(), EVM.SubState.t(), EVM.ExecEnv.t(), EVM.VM.output()}
+  def exp_mod(gas, exec_env) do
+    length_data = Memory.read_zeroed_memory(exec_env.data, 0, 96)
+
+    <<b_length_bin::binary-size(32), e_length_bin::binary-size(32),
+      m_length_bin::binary-size(32)>> = length_data
+
+    b_length = :binary.decode_unsigned(b_length_bin)
+    e_length = :binary.decode_unsigned(e_length_bin)
+    m_length = :binary.decode_unsigned(m_length_bin)
```

Comment:
What do you think about calling these `base_length`, `exponent_length` and `modulus_length` to more closely match what EIP 198 calls them? It may also make this code more clear to someone who looks at this code in the future without as much context.